subscriptions/test_tasks.py

```diff
@@ -103,3 +105,57 @@ def test_calculate_subscription_lifecycle_behind(self):
         )
         self.assertEqual(behind.expected_messageset, subscription.messageset)
         self.assertEqual(behind.expected_sequence_number, 3)
+
+
+class CachedMessageLookupTests(TestCase):
+    def test_caching_message_lookup_working(self):
+        """
+        Ensure that the second time we make a request, there's no database hit
+        """
+        with disable_signal("post_save", schedule_saved, Schedule):
+            schedule = Schedule.objects.create(minute=0)
+        messageset = MessageSet.objects.create(default_schedule=schedule)
+        for i in range(10):
+            Message.objects.create(
+                messageset=messageset, text_content=str(i), sequence_number=i
+            )
+        subscription_1 = Subscription.objects.create(
+            schedule=schedule, messageset=messageset
+        )
+        subscription_2 = Subscription.objects.create(
+            schedule=schedule, messageset=messageset
+        )
+        subscription_1.save()
+        subscription_2.save()
```

Comment:
We shouldn't need to call `.save()` here, as the `.create(..)` creates the object in the database.